packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -361,7 +365,15 @@ export function useAsyncData<
     error: writableComputedRef(() => nuxtApp._asyncData[key.value]?.error as Ref<NuxtErrorDataT extends Error | NuxtError ? NuxtErrorDataT : NuxtError<NuxtErrorDataT>>),
     refresh: (...args) => nuxtApp._asyncData[key.value]!.execute(...args),
     execute: (...args) => nuxtApp._asyncData[key.value]!.execute(...args),
-    clear: () => clearNuxtDataByKey(nuxtApp, key.value),
+    clear: () => {
+      if (options.abortController) {
+        options.abortController.abort(new DOMException('Request aborted as the async data was cleared.', 'AbortError'))
+      }
+      if (nuxtApp._asyncDataPromises[key.value]) {
+        (nuxtApp._asyncDataPromises[key.value] as any).cancelled = true
+      }
```

Comment:
I'm just asking, but don't we need something like this, too?

```ts
options.abortController.signal.addEventListener('abort', () => {
  if (nuxtApp._asyncDataPromises[key.value]) {
        (nuxtApp._asyncDataPromises[key.value] as any).cancelled = true
      }
})
```

What if someone cancels the signal from outside?

Reply:
make sense